nixos/modules/services/web-apps/lasuite-docs.nix

```diff
@@ -0,0 +1,576 @@
+    };
+
+    collaborationServer = {
+      package = mkPackageOption pkgs "lasuite-docs-collaboration-server" { };
+
+      port = mkOption {
+        type = types.port;
+        default = 4444;
+        description = ''
+          Port used by the collaboration server to listen.
+        '';
+      };
+
+      serverSecretPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the server secret of the collaboration server.
+        '';
+      };
+
+      yproviderApiKeyPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the y-provider API key.
+        '';
+      };
+
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = types.attrsOf (
+            types.oneOf [
+              types.str
+              types.bool
+            ]
+          );
+
+          options = {
+            PORT = mkOption {
+              type = types.str;
+              default = toString cfg.collaborationServer.port;
+              readOnly = true;
+              description = "Port used by collaboration server to listen to";
+            };
+
+            COLLABORATION_BACKEND_BASE_URL = mkOption {
+              type = types.str;
+              default = "https://${cfg.domain}";
+              defaultText = "https://\${cfg.domain}";
+              description = "URL to the backend server base";
+            };
+          };
+        };
+        default = { };
+        example = ''
+          {
+            COLLABORATION_LOGGING = true;
+          }
+        '';
+        description = ''
+          Configuration options of collaboration server.
+
+          See https://github.com/suitenumerique/docs/blob/v${cfg.collaborationServer.package.version}/docs/env.md
+        '';
+      };
+    };
+
+    gunicorn = {
+      extraArgs = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Extra arguments to pass to the gunicorn process.
+        '';
+      };
+    };
+
+    celery = {
+      extraArgs = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Extra arguments to pass to the celery process.
+        '';
+      };
+    };
+
+    domain = mkOption {
+      type = types.str;
+      description = ''
+        Domain name of the docs instance.
+      '';
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = types.attrsOf (
+          types.nullOr (
+            types.oneOf [
+              types.str
+              types.bool
+              types.path
+              types.int
+            ]
+          )
+        );
+
+        options = {
+          DJANGO_CONFIGURATION = mkOption {
+            type = types.str;
+            default = "Production";
+            description = "The configuration that Django will use";
+          };
```

Comment:
I would suggest making that option internal. This is nothing a production system should ever touch.